druid-shell/src/platform/windows/paint.rs

```diff
@@ -110,16 +146,19 @@ pub(crate) unsafe fn create_render_target_dxgi(
     }
 }
 
-impl<'a> PaintCtx<'a> {
-    /// Return the raw Direct2D factory for this painting context. Note: it's possible
-    /// this will be wrapped to make it easier to port.
-    pub fn d2d_factory(&self) -> &direct2d::Factory {
-        self.d2d_factory
-    }
 
-    /// Return the raw Direct2D RenderTarget for this painting context. Note: it's possible
-    /// this will be wrapped to make it easier to port.
-    pub fn render_target(&mut self) -> &mut GenericRenderTarget {
-        self.render_target
+/// Casts hwnd variant to DeviceTarget
+///
+/// TODO: investigate whether there's a better way to do this.
```

Comment:
Right, here I think you want a ComPtr cast, but this logic is *probably* correct.